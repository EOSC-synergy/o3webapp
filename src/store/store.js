import { applyMiddleware, combineReducers, configureStore, createStore } from '@reduxjs/toolkit';
import plotReducer from './plotSlice/plotSlice';
import modelsReducer, { setModelsOfModelGroup } from './modelsSlice/modelsSlice';
import referenceReducer from './referenceSlice/referenceSlice';
import apiReducer from '../services/API/apiSlice/apiSlice';
import { DEFAULT_MODEL_GROUP } from '../utils/constants';
import { HYDRATE, createWrapper } from 'next-redux-wrapper';
import thunkMiddleware from 'redux-thunk';

const bindMiddleware = (middleware) => {
    if (process.env.NODE_ENV !== 'production') {
        const { composeWithDevTools } = require('redux-devtools-extension');
        return composeWithDevTools(applyMiddleware(...middleware));
    }
    return applyMiddleware(...middleware);
};

const combinedReducer = combineReducers({
    plot: plotReducer,
    models: modelsReducer,
    reference: referenceReducer,
    api: apiReducer,
});

const reducer = (state, action) => {
    if (action.type === HYDRATE) {
        const nextState = {
            ...state, // use previous state
            ...action.payload, // apply delta from hydration
        };
        // if (state.count.count) nextState.count.count = state.count.count; // preserve count value on client side navigation
        return nextState;
    } else {
        return combinedReducer(state, action);
    }
};

let store;

const initStore = () => {
    store = createStore(reducer, bindMiddleware([thunkMiddleware]));
    return store;
};

export { store };

export const wrapper = createWrapper(initStore);

export function createTestStore() {
    const store = configureStore({
        reducer: combinedReducer,
    });
    store.dispatch(setModelsOfModelGroup(DEFAULT_MODEL_GROUP)); // fill with default group
    return store;
}
