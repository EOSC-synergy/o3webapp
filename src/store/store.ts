import { Action, combineReducers, configureStore, ThunkAction } from '@reduxjs/toolkit';
import plotReducer, { GlobalPlotState } from './plotSlice';
import modelsReducer, { GlobalModelState, setModelsOfModelGroup } from './modelsSlice';
import referenceReducer, { GlobalReferenceState } from './referenceSlice';
import apiReducer from '../services/API/apiSlice/apiSlice';
import { DEFAULT_MODEL_GROUP } from '../utils/constants';
import { createWrapper, HYDRATE } from 'next-redux-wrapper';
import thunkMiddleware from 'redux-thunk';
import { useDispatch, useStore } from 'react-redux';

type GlobalState =
    | undefined
    | (GlobalPlotState & GlobalModelState & GlobalReferenceState & { api: any });

const combinedReducer = combineReducers({
    plot: plotReducer,
    models: modelsReducer,
    reference: referenceReducer,
    api: apiReducer,
});

const reducer = (state: GlobalState, action: { type: string; payload: any }) => {
    if (action.type === HYDRATE) {
        // if (state.count.count) nextState.count.count = state.count.count; // preserve count value on client side navigation
        return {
            ...state, // use previous state
            ...action.payload, // apply delta from hydration
        };
    } else {
        return combinedReducer(state, action);
    }
};

//let store: ReturnType<typeof configureStore<GlobalState, >>;

const initStore = () => {
    //store = createStore(reducer, bindMiddleware([thunkMiddleware]));
    return configureStore({
        reducer,
        devTools: process.env.NODE_ENV !== 'production',
        middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(thunkMiddleware),
    });
};

export type AppStore = ReturnType<typeof initStore>;
export type AppState = ReturnType<AppStore['getState']>;
export type AppThunk<ReturnType = void> = ThunkAction<ReturnType, AppState, unknown, Action>;

export type AppDispatch = AppStore['dispatch'];
export const useAppDispatch = () => useDispatch<AppDispatch>(); // Export a hook that can be reused to resolve types

export const useAppStore = () => useStore<AppState>();

export const wrapper = createWrapper(initStore);

export function createTestStore() {
    const store = initStore();
    store.dispatch(setModelsOfModelGroup(DEFAULT_MODEL_GROUP)); // fill with default group
    return store;
}
