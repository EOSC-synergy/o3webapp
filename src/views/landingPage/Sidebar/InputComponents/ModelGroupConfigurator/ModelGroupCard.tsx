import React, { type ChangeEvent, type FC, useState } from 'react';
import Card from '@mui/material/Card';
import Typography from '@mui/material/Typography';
import VisibilityOffIcon from '@mui/icons-material/VisibilityOff';
import MuiVisibilityIcon from '@mui/icons-material/Visibility';
import DeleteIcon from '@mui/icons-material/Delete';
import { Checkbox, Divider, IconButton, FormControlLabel } from '@mui/material';
import Grid from '@mui/material/Grid';
import EditModelGroupModal from './EditModelGroupModal';
import AddModelGroupModal from './AddModelGroupModal';
import { useSelector } from 'react-redux';

import { CardActions, Button } from '@mui/material';
import {
    setStatisticalValueForGroup,
    selectStatisticalValueSettingsOfGroup,
    selectNameOfGroup,
    setVisibilityForGroup,
    selectVisibilityOfGroup,
    deleteModelGroup,
} from 'store/modelsSlice';
import { STATISTICAL_VALUES, STATISTICAL_VALUES_LIST } from 'utils/constants';
import { type AppState, useAppDispatch } from 'store';
import { type ErrorReporter } from 'utils/reportError';

type ModelGroupCardProps = {
    /**
     * The id of the model group this card should belong to
     */
    modelGroupId: number;
    /**
     * function for error handling
     */
    reportError: ErrorReporter;
};

/**
 * A card containing information about a model group.
 * Used in {@link ModelGroupConfigurator}.
 *
 * @component
 * @param {Object} props
 * @param {String} props.reportError - error function
 * @param {int} props.modelGroupId -> id of the model group
 * @returns {JSX.Element} a jsx containing a modal with a data grid with all models from the model group
 */
const ModelGroupCard: FC<ModelGroupCardProps> = (props) => {
    const dispatch = useAppDispatch();

    /**
     * the name of the model group. Retrieved from the Redux Store
     * @see {@link selectNameOfGroup}
     * @constant {Array}
     */
    const modelGroupName = useSelector((state: AppState) =>
        selectNameOfGroup(state, props.modelGroupId)
    );

    /**
     * gets the statisticalVales of the model Group. Retrieved from the Redux Store.
     * @see {@link selectStatisticalValueSettingsOfGroup}
     * @constant {Object}
     */
    const modelGroupStatisticalValue = useSelector((state: AppState) =>
        selectStatisticalValueSettingsOfGroup(state, props.modelGroupId)
    );

    /**
     * whether the model group is currenly visible or not. Retrieved from the Redux Store.
     * @see {@link selectVisibilityOfGroup}
     * @constant {boolean}
     */
    const isModelGroupVisible = useSelector((state: AppState) =>
        selectVisibilityOfGroup(state, props.modelGroupId)
    );

    /**
     * Toggles the visibility of the given statistical value
     * by dispatching an action to the redux store.
     *
     * @param {Event} event the event that called this function
     * @param {String} statisticalValue the name of the statistical value that should be toggled
     * @function
     */
    const toggleModelGroupStatisticalValueVisibility = (
        event: ChangeEvent<HTMLInputElement>,
        statisticalValue: STATISTICAL_VALUES
    ) => {
        dispatch(
            setStatisticalValueForGroup({
                groupId: props.modelGroupId,
                svType: statisticalValue,
                isIncluded: event.target.checked,
            })
        );
    };

    /**
     * Toggles the visibility of the whole group in the graph
     * by dispatching an action to the redux store.
     * @function
     */
    const toggleModelGroupVisibility = () => {
        dispatch(
            setVisibilityForGroup({ groupId: props.modelGroupId, isVisible: !isModelGroupVisible })
        );
        /*
        // This de-/activates all statistical values when toggling the group visibility.
        for (const key in STATISTICAL_VALUES) {
            dispatch(setStatisticalValueForGroup(
                {groupId: props.modelGroupId, svType: key, isIncluded: !isModelGroupVisible}
            ));
        }
         */
    };

    /**
     * State to keep track of whether the edit group modal is currently visible or not.
     * @constant {Array}
     * @default false
     */
    const [isEditModalVisible, setEditModalVisible] = useState(false);

    /**
     * State to keep track of whether the add group modal (used to edit group members) is currently visible or not.
     * @constant {Array}
     * @default false
     */
    const [isAddModalVisible, setAddModalVisible] = useState(false);

    /**
     * State to keep track of whether a delete request is open or not.
     * @constant {Array}
     * @default false
     */
    const [isDeleteRequest, setDeleteRequest] = useState(false);

    const [refreshAddModelGroupModalState, setRefreshAddModelGroupModalState] = useState(true);
    const [refreshEditModalGroupModalState, setRefreshEditModalGroupModalState] = useState(true);

    /**
     * Shows the edit group modal.
     * @cfunction
     */
    const showEditModal = (refresh: boolean) => {
        setAddModalVisible(false); // avoid two modals being visible under all circumstances
        setEditModalVisible(true);
        setRefreshEditModalGroupModalState(refresh);
    };

    /**
     * Closes the edit modal.
     * @function
     */
    const closeEditModal = () => {
        setEditModalVisible(false);
    };

    /**
     * Shows the add model group modal (used to edit group members).
     * @function
     */
    const showAddModal = (refresh: boolean) => {
        setEditModalVisible(false); // avoid two modals being visible under all circumstances
        setAddModalVisible(true);
        setRefreshAddModelGroupModalState(refresh);
    };

    /**
     * Closes the add model group modal (used to edit group members).
     * @function
     */
    const closeAddModal = () => {
        setAddModalVisible(false);
    };

    /**
     * Returns the visibility icon, depending on whether the group is currently visible or not.
     * @function
     * @returns {JSX.Element} JSX containing an icon of an open or closed eye depending on isModelGroupVisible
     */
    const VisibilityIcon = () => {
        return isModelGroupVisible ? (
            <MuiVisibilityIcon data-testid="ModelGroupCard-VisibilityIcon-visible" />
        ) : (
            <VisibilityOffIcon data-testid="ModelGroupCard-VisibilityIcon-invisible" />
        );
    };

    /**
     * Toggles the visibility of the delete request card.
     * Always sets it to the opposite of what it is now.
     * @function
     */
    const toggleDeleteRequest = () => {
        setDeleteRequest(!isDeleteRequest);
    };

    /**
     * Deletes a model group.
     * Dispatches the deletion to Redux with the group id of the model group card.
     * @function
     */
    const deleteGroup = () => {
        toggleDeleteRequest();
        dispatch(deleteModelGroup({ groupId: props.modelGroupId }));
    };

    if (!isDeleteRequest) {
        return (
            <Card
                style={{ margin: '3%', padding: '2%', width: '300px', height: '210px' }}
                elevation={2}
            >
                <EditModelGroupModal
                    modelGroupId={props.modelGroupId}
                    isOpen={isEditModalVisible}
                    onClose={closeEditModal}
                    setOpen={showEditModal}
                    refresh={refreshEditModalGroupModalState}
                />
                <AddModelGroupModal
                    modelGroupId={props.modelGroupId}
                    isOpen={isAddModalVisible}
                    onClose={closeAddModal}
                    reportError={props.reportError}
                    setOpen={showAddModal}
                    refresh={refreshAddModelGroupModalState}
                />
                <Grid container>
                    <Grid item xs={2}>
                        <IconButton
                            aria-label="change visibility"
                            onClick={toggleModelGroupVisibility}
                        >
                            <VisibilityIcon />
                        </IconButton>
                    </Grid>
                    <Grid item xs={8} textAlign="center">
                        <Typography variant="h6" data-testid="ModelGroupCard-groupName">
                            {modelGroupName}
                        </Typography>
                    </Grid>
                    <Grid item xs={2}>
                        <IconButton
                            aria-label="delete model group"
                            onClick={toggleDeleteRequest}
                            data-testid="ModelGroupCard-delete-model-group"
                        >
                            <DeleteIcon />
                        </IconButton>
                    </Grid>
                </Grid>
                <Divider />
                <Grid container sx={{ paddingTop: '0.5em' }}>
                    {STATISTICAL_VALUES_LIST.map((key, idx) => {
                        return (
                            <Grid item key={idx} xs={6} sx={{ paddingLeft: '1em' }}>
                                <FormControlLabel
                                    label={key}
                                    id={`ModelGroupCard-toggle-${key}-label`}
                                    control={
                                        <Checkbox
                                            onChange={(event) =>
                                                toggleModelGroupStatisticalValueVisibility(
                                                    event,
                                                    key
                                                )
                                            }
                                            checked={modelGroupStatisticalValue[key]}
                                            id={`ModelGroupCard-toggle-${key}-checkbox`}
                                            //labelid={`ModelGroupCard-toggle-${key}-label`}
                                            data-testid={`ModelGroupCard-toggle-${key}-checkbox`}
                                        />
                                    }
                                />
                            </Grid>
                        );
                    })}
                </Grid>
                <Divider />
                <CardActions>
                    <Button
                        size="small"
                        variant="outlined"
                        onClick={() => showEditModal(true)}
                        data-testid="ModelGroupCard-EditModelGroupModal-button-open"
                    >
                        Edit statistical values
                    </Button>
                    <Button
                        size="small"
                        variant="outlined"
                        onClick={() => showAddModal(true)}
                        data-testid="ModelGroupCard-AddModelGroupModal-button-open"
                    >
                        Edit group members
                    </Button>
                </CardActions>
            </Card>
        );
    } else {
        return (
            <Card
                style={{
                    margin: '3%',
                    padding: '2%',
                    width: '300px',
                    height: '210px',
                }}
                elevation={2}
            >
                <Grid container sx={{ mt: '60px' }}>
                    <Grid item xs={12} sx={{ mb: '5px', textAlign: 'center' }}>
                        <Typography variant="h6">Delete this model group?</Typography>
                    </Grid>
                    <Grid item xs={12} sx={{ mt: '5px', textAlign: 'center' }}>
                        <Button
                            size="medium"
                            variant="outlined"
                            onClick={toggleDeleteRequest}
                            sx={{
                                mr: '10px',
                            }}
                            data-testid="ModelGroupCard-delete-model-keep"
                        >
                            Keep
                        </Button>
                        <Button
                            size="medium"
                            //variant="filled"
                            onClick={deleteGroup}
                            sx={{
                                ml: '10px',
                                color: 'black',
                                backgroundColor: '#fed136',
                                '&:hover': {
                                    backgroundColor: '#b19225',
                                },
                            }}
                            data-testid="ModelGroupCard-delete-model-delete"
                        >
                            Delete
                        </Button>
                    </Grid>
                </Grid>
            </Card>
        );
    }
};

export default ModelGroupCard;
