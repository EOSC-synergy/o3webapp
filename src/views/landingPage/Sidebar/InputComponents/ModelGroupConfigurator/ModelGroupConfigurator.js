import React from "react";
import AddModelGroupModal from "./AddModelGroupModal/AddModelGroupModal";
import ModelGroupCard from "./ModelGroupCard/ModelGroupCard";
import { Button } from '@mui/material';
import { useSelector } from "react-redux"
import PropTypes from 'prop-types';
import { selectAllGroupIds } from "../../../../../store/modelsSlice/modelsSlice";

/**
 * enables the user to configure models that should be visible in the plot clustered as model groups
 * @component
 * @param {Object} props 
 * @param {function} props.reportError - function to report errors
 * @returns {JSX} a jsx containing a ModelGroupModal and a ModelGroupCard and EditModelGroupModal per model group
 */
function ModelGroupConfigurator(props) {

    const allGroupIds = useSelector(selectAllGroupIds);


    const [isAddModalVisible, setAddModalVisible] = React.useState(false);

    const closeAddModal = () => {
        setAddModalVisible(false);
    }

    const openAddModal = () => {
        setAddModalVisible(true);
    }

    return (
        <>
            {
                allGroupIds.map((id, idx) => {
                    return (<ModelGroupCard key={idx} modelGroupId={id} reportError={props.reportError} />);
                })
            }
            <Button
                sx={{width: "100%"}}
                variant="contained"
                onClick={openAddModal}
                data-testid="ModelGroupConfigurator-addModelGroup-button"
            >
                Add Model Group
            </Button>
            <AddModelGroupModal isOpen={isAddModalVisible} onClose={closeAddModal} reportError={props.reportError} />
        </>
    );
}

ModelGroupConfigurator.propTypes = {
    reportError: PropTypes.func
}

export default ModelGroupConfigurator;