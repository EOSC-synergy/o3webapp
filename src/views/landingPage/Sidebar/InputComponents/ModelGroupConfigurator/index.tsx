import React, { type FC, useState } from 'react';
import AddModelGroupModal from './AddModelGroupModal';
import ModelGroupCard from './ModelGroupCard';
import { Button } from '@mui/material';
import { useSelector } from 'react-redux';
import { selectAllGroupIds } from 'store/modelsSlice';
import { type ErrorReporter } from 'utils/reportError';

type ModelGroupConfiguratorProps = {
    reportError: ErrorReporter;
};
/**
 * Enables the user to configure models that should be visible in the plot clustered as model
 * groups. Compromised of {@link EditModelGroupModal}, {@link ModelGroupCard} and
 * {@link AddModelGroupModal}.
 */
const ModelGroupConfigurator: FC<ModelGroupConfiguratorProps> = ({ reportError }) => {
    /**
     * Label that is displayed in the add model group button
     *
     * @constant {String}
     * @default 'Add Model Group'
     */
    const addModelGroupButtonLabel = 'Add Model Group';

    /**
     * Ids of all existing modelGroups
     *
     * @constant {Array}
     * @see {@link selectAllGroupIds}
     */
    const allGroupIds = useSelector(selectAllGroupIds);

    /**
     * State that tracks whether the addModelGroupModal is visible or not
     *
     * @constant {Array}
     */
    const [isAddModalVisible, setAddModalVisible] = useState(false);
    const [refreshState, setRefreshState] = useState(true);

    /**
     * Function to close addModelGroupModal
     *
     * @function
     */
    const closeAddModal = () => {
        setAddModalVisible(false);
    };

    /**
     * Function to open addModelGroupModal
     *
     * @function
     */
    const openAddModal = (refresh: boolean) => {
        setAddModalVisible(true);
        setRefreshState(refresh);
    };

    return (
        <>
            {allGroupIds.map((id, idx) => {
                return <ModelGroupCard key={idx} modelGroupId={id} reportError={reportError} />;
            })}
            <Button
                sx={{ width: '100%' }}
                variant="contained"
                onClick={() => openAddModal(true)}
                data-testid="ModelGroupConfigurator-addModelGroup-button"
            >
                {addModelGroupButtonLabel}
            </Button>
            <AddModelGroupModal
                isOpen={isAddModalVisible}
                onClose={closeAddModal}
                setOpen={openAddModal}
                refresh={refreshState}
                reportError={reportError}
            />
        </>
    );
};

export default ModelGroupConfigurator;
