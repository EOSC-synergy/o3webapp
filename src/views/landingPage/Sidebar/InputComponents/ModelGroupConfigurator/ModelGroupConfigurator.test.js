import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { createTestStore } from '../../../../../store/store';
import { Provider } from 'react-redux';
import ModelGroupConfigurator from './ModelGroupConfigurator';
import userEvent from '@testing-library/user-event';
import * as redux from 'react-redux';

let store;
describe('test ModelGroupCard rendering', () => {
    beforeEach(() => {
        store = createTestStore();
    });

    it('renders without crashing', () => {
        render(
            <Provider store={store}>
                <ModelGroupConfigurator reportError={jest.fn()} />
            </Provider>
        );
    });

    it('renders correctly', () => {
        const { container } = render(
            <Provider store={store}>
                <ModelGroupConfigurator reportError={jest.fn()} />
            </Provider>
        );
        expect(container).toMatchSnapshot();
    });

    it('renders add model group button', () => {
        const { getByTestId } = render(
            <Provider store={store}>
                <ModelGroupConfigurator reportError={jest.fn()} />
            </Provider>
        );
        expect(getByTestId(/addModelGroup-button/)).toBeInTheDocument();
        expect(getByTestId(/addModelGroup-button/)).toBeEnabled();
    });

    it('raises a console.error function if a required prop is not provided', () => {
        console.error = jest.fn();
        render(
            <Provider store={store}>
                <ModelGroupConfigurator />
            </Provider>
        );
        expect(console.error).toHaveBeenCalled();
    });
});

describe('test functionality buttons', () => {
    beforeEach(() => {
        store = createTestStore();
    });

    it('renders add model group modal', () => {
        const { getByTestId } = render(
            <Provider store={store}>
                <ModelGroupConfigurator reportError={jest.fn()} />
            </Provider>
        );
        userEvent.click(getByTestId(/addModelGroup-button/));
        expect(getByTestId(/addModelGroupModal/)).toBeInTheDocument();
    });
});

describe('test functionality redux', () => {
    beforeEach(() => {
        store = createTestStore();
    });

    it('renders all model group cards in store', () => {
        const spy = jest.spyOn(redux, 'useSelector');
        const modelGroupName = 'blob';
        const fakeModelGroupName = 'blub';

        spy.mockReturnValueOnce([0])
            .mockReturnValueOnce(modelGroupName)
            .mockReturnValueOnce({
                median: true,
                mean: true,
                'standard deviation': false,
                percentile: true,
            })
            .mockReturnValueOnce(true);

        const { getByText, queryByText } = render(
            <Provider store={store}>
                <ModelGroupConfigurator reportError={jest.fn()} />
            </Provider>
        );

        expect(getByText(modelGroupName)).toBeInTheDocument();
        expect(queryByText(fakeModelGroupName)).toBeNull();
    });
});
