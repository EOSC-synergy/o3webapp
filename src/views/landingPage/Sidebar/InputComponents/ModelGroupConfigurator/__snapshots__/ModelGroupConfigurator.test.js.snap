// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ModelGroupCard rendering renders correctly 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 MuiCard-root css-935hc6-MuiPaper-root-MuiCard-root"
    style="margin: 3%; padding: 2%; width: 300px; height: 210px;"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
      >
        <button
          aria-label="change visibility"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ModelGroupCard-VisibilityIcon-visible"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-epm2de-MuiGrid-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          data-testid="ModelGroupCard-groupName"
        >
          Example Group
        </h6>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
      >
        <button
          aria-label="delete model group"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="DeleteIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiGrid-root MuiGrid-container css-m6sf28-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-d83jk5-MuiGrid-root"
      >
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          id="ModelGroupCard-toggle-mean-label"
        >
          <span
            class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            data-testid="ModelGroupCard-toggle-mean-checkbox"
            labelid="ModelGroupCard-toggle-mean-label"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              id="ModelGroupCard-toggle-mean-checkbox"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
          >
            mean
          </span>
        </label>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-d83jk5-MuiGrid-root"
      >
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          id="ModelGroupCard-toggle-standard deviation-label"
        >
          <span
            class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            data-testid="ModelGroupCard-toggle-standard deviation-checkbox"
            labelid="ModelGroupCard-toggle-standard deviation-label"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              id="ModelGroupCard-toggle-standard deviation-checkbox"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
          >
            standard deviation
          </span>
        </label>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-d83jk5-MuiGrid-root"
      >
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          id="ModelGroupCard-toggle-median-label"
        >
          <span
            class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            data-testid="ModelGroupCard-toggle-median-checkbox"
            labelid="ModelGroupCard-toggle-median-label"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              id="ModelGroupCard-toggle-median-checkbox"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
          >
            median
          </span>
        </label>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-d83jk5-MuiGrid-root"
      >
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          id="ModelGroupCard-toggle-percentile-label"
        >
          <span
            class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            data-testid="ModelGroupCard-toggle-percentile-checkbox"
            labelid="ModelGroupCard-toggle-percentile-label"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              id="ModelGroupCard-toggle-percentile-checkbox"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
          >
            percentile
          </span>
        </label>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
    >
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1k23hlb-MuiButtonBase-root-MuiButton-root"
        data-testid="ModelGroupCard-EditModelGroupModal-button-open"
        tabindex="0"
        type="button"
      >
        Edit statistical values
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1k23hlb-MuiButtonBase-root-MuiButton-root"
        data-testid="ModelGroupCard-AddModelGroupModal-button-open"
        tabindex="0"
        type="button"
      >
        Edit group members
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
    data-testid="ModelGroupConfigurator-addModelGroup-button"
    tabindex="0"
    type="button"
  >
    Add Model Group
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
